### PR TITLE
Add support for | operator between StrawberryUnion and None

### PR DESCRIPTION
**Pull request by Apakottur**
This is an attempt to fix https://github.com/strawberry-graphql/strawberry/issues/1539.

Adds support for annotating unions are optional with the new OR notation.

Note that we are also interested in merging unions at some point (see https://github.com/strawberry-graphql/strawberry/issues/711), but in this PR we only handle a union with `None`.

## Types of Changes

- [ ] Core
- [ ] Bugfix
- [ ] New feature
- [ ] Enhancement/optimization
- [ ] Documentation

## Issues Fixed or Closed by This PR

Closes https://github.com/strawberry-graphql/strawberry/issues/1539

## Checklist

- [ ] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [ ] I have read the CONTRIBUTING document.
- [ ] I have added tests to cover my changes.
- [ ] I have tested the changes and verified that they work and don't break anything (as well as I can manage).

### CODE REVIEW

**Comment by botberry** [comment]
Hi, thanks for contributing to Strawberry 🍓!

We noticed that this PR is missing a `RELEASE.md` file. We use that to automatically do releases here on GitHub and, most importantly, to PyPI!

So as soon as this PR is merged, a release will be made 🚀.

Here's an example of `RELEASE.md`:

```markdown
Release type: patch

Description of the changes, ideally with some examples, if adding a new feature.
```

Release type can be one of patch, minor or major. We use [semver](https://semver.org/), so make sure to pick the appropriate type. If in doubt feel free to ask :)


Here's the tweet text: 
```
🆕 Release (next) is out! Thanks to Yossi Rozantsev for the PR 👏

Get it here 👉 https://github.com/strawberry-graphql/strawberry/releases/tag/(next)
```

**Comment by patrick91** [comment]
@Apakottur thanks! I've already a PR for this though: https://github.com/strawberry-graphql/strawberry/pull/1455

Would you be ok in checking that one and see if covers your use case? :)

**Comment by Apakottur** [comment]
@patrick91 Cool, I missed that PR when I was looking for similar fixes.

We are using https://github.com/asottile/pyupgrade which automatically converts all unions in types to the new notation, which breaks until we merge  the fix in this PR (or in your PR). Any chance we can handle the `None` case separately to support the syntax in a minimal way, without having to pressure/wait for the other merging logic?

**Comment by patrick91** [comment]
@Apakottur sure, we can merge this PR when you add the release files.

I think I want to change how union works anyway :)

**Comment by Apakottur** [comment]
@patrick91 Thanks.

I merged our implementations into one, and added the annotation unit tests from your PR (alongside the schema one from mine). I think this should lead to less conflicts when you proceed with your PR :)

**Review by patrick91 — APPROVED** [review]

**Pull request merged**
